  <md-card class="md-padding docs">


    <h2 class="">Global Alliance Genomics API</h2>

    <p class=" md-body-3 text-justify">
        The Global Alliance for Genomics and Health (GA4GH) Genomics
        <a href="/#/api/v0.5.1/"><strong> v0.5.1 <i class="fa fa-file-code-o"></i>
        </strong> API</a> will allow the interoperable exchange of genomic
        information across multiple organizations and on multiple
        platforms.
    
        This is a freely available open standard for interoperability,
        that uses common web protocols to support serving and sharing
        of data on DNA sequences and genomic variation.
    
        The API is implemented as a webservice to create a data source
        which may be integrated into visualization software, web-based
        genomics portals or processed as part of genomic analysis
        pipelines.
    
        It overcomes the barriers of incompatible infrastructure
        between organizations and institutions to enable DNA data
        providers and consumers to better share genomic data and work
        together on a global scale, advancing genome research and
        clinical application.</p>

        <img  src="assets/images/documentation/GA4GH_API_interop.png">

      <p class="md-body-3 text-justify">
          This <a href="/#/api/v0.5.1"><strong>v0.5.1 <i
              class="fa fa-file-code-o"></i></strong></a>
          of the genomics API has been developed using Apache Avro to model the data. The documentation
          presented here is provided as a freely available blueprint for software developers to take and
          implement alongside those already at the EBI, NCBI and Google.
      </p>


    <p class="md-body-3 text-left text-justify">The Avro data model comprises of 3 main resources, Reads, Variants and the Reference. The following section describes each
      of these aspects of the model in turn, followed by an example of using the API to search for Variants</p>

<h2>Reads</h2>

<img class="lazy" data-original="assets/images/documentation/reads.png">

<ul class="md-body-3 text-left">
    <li>A <strong>GAReadGroupSet</strong> is a logical collection of <strong>GAReadGroups</strong>.
    </li>
    <li>A <strong>GAReadGroup</strong> is all the data that's processed the same way by the sequencer.
        There are typically 1-10 <strong>GAReadGroup's</strong> in a <strong>GAReadGroupSet</strong>.
    </li>
    <li>A <em>fragment</em> is a single stretch of a DNA molecule. There are typically
        millions of fragments in a <strong>GAReadGroup</strong>. A fragment has a name (QNAME in BAM
        spec), a length (TLEN in BAM spec), and an array of reads.
    </li>
    <li>A <em>read</em> is a contiguous sequence of bases. There are typically only one or
        two reads in a fragment. If there are two reads, they're known as a mate pair.
        A read has an array of base values, an array of base qualities, and alignment
        information.
    </li>
    <li>An <em>alignment</em> is the way alignment software maps a read to a reference.
        There's one primary alignment, and can be one or more secondary alignments.
        Secondary alignments represent alternate possible mappings.
    </li>
    <li>A <em>linear alignment</em> maps a string of bases to a reference using a single
        CIGAR string. There's one representative alignment, and can be one or more
        supplementary alignments. Supplementary alignments represent linear alignments
        that are subsets of a chimeric alignment.
    </li>
    <li>A <strong>GAReadAlignment</strong> object is a flattened representation of the bottom layers
        of this hierarchy. There's exactly one such object per <em>linear alignment</em>.
        The object contains alignment info, plus fragment and read info for easy access.
    </li>
</ul>

<h2>Variants</h2>

<div class="border"></div>
<img class="lazy" data-original="assets/images/documentation/variants.png">

<p class="md-body-3 text-left text-justify">A GAVariant represents a change in DNA sequence relative to some
    reference. For example, a variant could represent a SNP or an insertion. Variants belong to a
    GAVariantSet. This is equivalent to a row in VCF.</p>
<ul class="md-body-3 text-left">
    <li>
        <strong>VariantSet</strong> is all the variant information on a specified population
    </li>
    <li><strong>Variant</strong> (row) is a single variant (e.g. "a change A to T at offset 5,227,002 on
        chr 11", which causes sickle cell anemia)
    </li>
    <li><strong>CallSet</strong> (column) is the genotype information for a single tissue sample</li>
    <li><strong>Call</strong> (cell) is the data about a particular sample having a particular variant
        (e.g. "sample 31415 is 82% likely to have this variant")
    </li>
</ul>

<h2>Reference</h2>
<img class="lazy" data-original="assets/images/documentation/reference.png">
<ul class="md-body-3 text-left">
    <li>A <strong>GAReferenceSet</strong> is a logical collection of <strong>GAReferences</strong>.</li>
    <li>A <strong>GAReference</strong> represents an assembled consensus Genomic sequence and is comprised of a string of DNA bases.</li>
</ul>
    <h2>Example Usage</h2>
<div class="border"></div>
    <h3>Variants: search</h3>

<div class="text-left">
    <br>
    <p class="md-body-3">    When implemented as a REST web service the API may be used in the standard POST GET http request pattern. Below is an example of making a POST <a href="http://ga4gh.org/ga4gh_api.html#/schema/org.ga4gh.GASearchVariantsRequest">GASearchVariantsRequest</a> 
    request to query the <code>/variants/search</code> endpoint which returns a list of <a href="http://ga4gh.org/ga4gh_api.html#/schema/org.ga4gh.GAVariant">GAVariants</a> 
    matching the search criteria.</p>

    </p>
    <h3>Request</h3>
    <h4>HTTP request</h4>
    <pre>POST /variants/search</pre>

    <h4>Request body</h4>

    <p class="md-body-3">In the request body, supply data with the following structure as JSON:</p>
                <pre>
{
  "datasetId": string,
  "variantName": string,
  "callsetNames": [
    string
  ],
  "callsetIds": [
    string
  ],
  "contig": string,
  "startPosition": long,
  "endPosition": long,
  "pageToken": string,
  "maxResults": long
}
                </pre>


    <table class="table table-striped table-hover">
        <thead>
        <tr>
            <th>Type</th>
            <th>Field</th>
            <th>Default Value</th>
            <th>Description</th>
        </tr>
        </thead>
        <tbody>
        <tr>
            <td class="type">array&lt;string&gt;</td>
            <td class="field">variantSetIds</td>
            <td class="field-doc">[]</td>
            <td class="field-doc"><p>Required. The IDs of the variant sets to search over.</p></td>
        </tr>
        <tr>
            <td class="type">null&nbsp;|&nbsp;string</td>
            <td class="field">variantName</td>
            <td class="field-doc">null</td>
            <td class="field-doc"><p>Only return variants which have exactly this name.</p></td>
        </tr>
        <tr>
            <td class="type">array&lt;string&gt;</td>
            <td class="field">callSetIds</td>
            <td class="field-doc">[]</td>
            <td class="field-doc"><p>Only return variant calls which belong to call sets with these
                IDs.
                Leaving this blank returns all variant calls.</p></td>
        </tr>
        <tr>
            <td class="type">string</td>
            <td class="field">referenceName</td>
            <td class="field-doc"></td>
            <td class="field-doc"><p>Required. Only return variants on this reference.</p></td>
        </tr>
        <tr>
            <td class="type">long</td>
            <td class="field">start</td>
            <td class="field-doc"></td>
            <td class="field-doc"><p>Required. The beginning of the window (0-based, inclusive) for
                which overlapping variants should be returned.</p></td>
        </tr>
        <tr>
            <td class="type">long</td>
            <td class="field">end</td>
            <td class="field-doc"></td>
            <td class="field-doc"><p>Required. The end of the window (0-based, exclusive) for which
                overlapping
                variants should be returned.</p></td>
        </tr>
        <tr>
            <td class="type">null&nbsp;|&nbsp;string</td>
            <td class="field">pageToken</td>
            <td class="field-doc">null</td>
            <td class="field-doc"><p>The continuation token, which is used to page through large
                result sets.
                To get the next page of results, set this parameter to the value of
                <code>nextPageToken</code> from the previous response.</p></td>
        </tr>
        <tr>
            <td class="type">long</td>
            <td class="field">maxResults</td>
            <td class="field-doc">10</td>
            <td class="field-doc"><p>The maximum number of variants to return in each response.
                If more variants match this request, the <code>pageToken</code> can be used to
                fetch the next page of responses.</p></td>
        </tr>
        </tbody>
    </table>

    <h3>Response</h3>

    <p class="md-body-3">If successful, this method returns a response body as a <a href="http://ga4gh.org/ga4gh_api.html#/schema/org.ga4gh.GASearchVariantsResponse">GASearchVariantsResponse</a> as JSON with the following structure:</p>
                  <pre>
{
  "variants": [
    {
      "datasetId": string,
      "id": string,
      "names": [
        string
      ],
      "created": long,
      "contig": string,
      "position": long,
      "referenceBases": string,
      "alternateBases": [
        string
      ],
      "info": {
        (key): [
          string
        ]
      },
      "calls": [
        {
          "callsetId": string,
          "callsetName": string,
          "genotype": [
            integer
          ],
          "phaseset": string,
          "genotypeLikelihood": [
            double
          ],
          "info": {
            (key): [
              string
            ]
          }
        }
      ]
    }
  ],
  "nextPageToken": string
}
            </pre>


    <table class="table table-striped table-hover">
        <thead>
        <tr>
            <th>Type</th>
            <th>Field</th>
            <th>Default Value</th>
            <th>Description</th>
        </tr>
        </thead>
        <tbody>
        <tr>
            <td class="type">array&lt;<a
                    href="http://ga4gh.org/ga4gh_api.html#/schema/org.ga4gh.GAVariant"
                    data-original-title="" style="width: 600px;">GAVariant</a>&gt;</td>
            <td class="field">variants</td>
            <td class="field-doc">[]</td>
            <td class="field-doc"><p>The list of matching variants.
                If the <code>callSetId</code> field on the returned calls is not present,
                the ordering of the call sets from a <code>SearchCallSetsRequest</code>
                over the parent <code>GAVariantSet</code> is guaranteed to match the ordering
                of the calls on each <code>GAVariant</code>. The number of results will also be
                the same.</p></td>
        </tr>
        <tr>
            <td class="type">null&nbsp;|&nbsp;string</td>
            <td class="field">nextPageToken</td>
            <td class="field-doc">null</td>
            <td class="field-doc"><p>The continuation token, which is used to page through large
                result sets.
                Provide this value in a subsequent request to return the next page of
                results. This field will be empty if there aren't any additional results.</p>
            </td>
        </tr>
        </tbody>
    </table>
</div>

<p class="md-body-3">A full list of methods and endpoints can be found here 
<a href="/#/api/v0.5.1"><strong>on our v0.5.1 documentation page <i class="fa fa-file-code-o"></i></p>



<!-- <div ng-include src="'partials/footer.html'"></div> -->
</md-card>
